### PR TITLE
Update CHANGELOG with post 0.3.7 changes

### PR DESCRIPTION
**Pull request by catsby**
When through the PRs list and added changes / fixes that weren't themselves introduced post 0.3.7

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -61,6 +68,14 @@ BUG FIXES:
   * providers/aws: `source_dest_check` regression fixed (now works). [GH-1020]
   * providers/aws: Longer wait times for DB instances.
   * providers/aws: Longer wait times for route53 records (30 mins). [GH-1164]
+  * providers/aws: Fix support for TXT records in Route 53. [GH-1213]
+  * providers/aws: Fix support for wildcard records in Route 53. [GH-1222]
+  * providers/aws: Fix issue with ignoring the 'self' attribute of a Security Group rule. [GH-1223]
+  * providers/aws: Fix issue with `sql_mode` in RDS parameter group always causing an update. [GH-1225]
+  * providers/aws: Fix dependency violation with subnets and security groups [GH-1252]
+  * providers/aws: Fix issue with refreshing `db_subnet_groups` causing an error instead of updating state [GH-1254]
```
Comment (mitchellh) [inline CHANGELOG.md]: Don't go past 80 chars, wrap like some other lines here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -43,6 +45,11 @@ IMPROVEMENTS:
   * command/remote-config: After enabling remote state, a `pull` is
       automatically done initially.
   * providers/google: Add `size` option to disk blocks for instances. [GH-1284]
+  * providers/aws: Improve support for tagging resources.
+  * providers/aws: Add a short syntax for Route 53 Record names, e.g. `www` instead of 
+    `www.example.com`.
```
Comment (mitchellh) [inline CHANGELOG.md]: When I wrap I indent the 2nd+ lines with a couple more spaces, see above.

File: CHANGELOG.md [hunk 3]
```diff
@@ -23,6 +24,8 @@ FEATURES:
       or system killing Terraform.
   * **Math operations** in interpolations. You can now do things like
       `${count.index+1}`. [GH-1068]
+  * **New AWS SDK:** Move to `aws-sdk-go` (hashicorp/aws-sdk-go), 
+      a fork of the offical `awslabs` repo.
```
Comment (phinze) [inline CHANGELOG.md]: Might want to mention intent to get in sync with upstream so as not to cause any undue distress.

**Comment by phinze** [comment]
LGTM

**Pull request merged**